op-service/txmgr/txmgr_test.go

```diff
@@ -1003,3 +1003,61 @@ func TestNonceReset(t *testing.T) {
 	// internal nonce tracking should be reset every 3rd tx
 	require.Equal(t, []uint64{0, 0, 1, 2, 0, 1, 2, 0}, nonces)
 }
+
+// TestClose ensures that the tx manager will refuse new work and cancel any in progress
+func TestClose(t *testing.T) {
+	conf := configWithNumConfs(1)
+	conf.SafeAbortNonceTooLowCount = 100
+	h := newTestHarnessWithConfig(t, conf)
+
+	// make the tx fail until it tries `ready` number of times
+	// as currently set, this will fail 3 times before succeeding, delaying for 30ms total
+	i := 0
+	ready := 3
+	called := 0
+	sendTx := func(ctx context.Context, tx *types.Transaction) error {
+		defer func() { i = (i + 1) % (ready + 1) }()
+		defer func() { called += 1 }()
+		if i == ready {
+			txHash := tx.Hash()
+			h.backend.mine(&txHash, tx.GasFeeCap())
+			return nil
+		} else {
+			time.Sleep(10 * time.Millisecond)
+			return core.ErrNonceTooLow
+		}
+	}
```

Comment:
maybe easier to read
```suggestion
	sendTx := func(ctx context.Context, tx *types.Transaction) (err error) {
		if (called+1) % ready == 0 { // double check that this makes sense
			txHash := tx.Hash()
			h.backend.mine(&txHash, tx.GasFeeCap())
		} else {
			time.Sleep(10 * time.Millisecond)
			err = core.ErrNonceTooLow
		}
		called += 1
		return err
	}
```
and can remove `i`. should then also set `const retry = 4`. and we can then reuse it below in test assertions where you use `4`.

Reply:
Slightly modified this, but great suggestion, tyvm.
(incrementing `called` at the start makes the math simpler)

Reply:
even nicer!